src/renderer/components/+config-limit-ranges/limit-range-details.tsx

```diff
@@ -0,0 +1,134 @@
+import { KubeObjectDetailsProps } from "../kube-object";
+import { ILimitRangeItem, LimitRange, Resource } from "../../api/endpoints/limit-range.api";
+import { kubeObjectDetailRegistry } from "../../api/kube-object-detail-registry";
+import { KubeObjectMeta } from "../kube-object/kube-object-meta";
+import { DrawerItem } from "../drawer/drawer-item";
+import { Badge } from "../badge";
+
+interface Props extends KubeObjectDetailsProps<LimitRange> {
+}
+
+function renderLimits(limit: ILimitRangeItem, resource: string): JSX.Element {
+
+  return (
+    <div>
+      {limit.defaultRequest &&
+      Object.keys(limit.defaultRequest).filter(key => key === resource).map(resourceName => (
+        <Badge key={resourceName} label={`defaultRequest: ${limit.defaultRequest[resourceName]}`}/>
+      ))
+      }
+      {limit.default &&
+      Object.keys(limit.default).filter(key => key === resource).map(resourceName => (
+        <Badge key={resourceName} label={`default: ${limit.default[resourceName]}`}/>
+      ))
+      }
+      {limit.maxLimitRequestRatio &&
+      Object.keys(limit.maxLimitRequestRatio).filter(key => key === resource).map(resourceName => (
+        <Badge key={resourceName} label={`maxLimitRequestRatio: ${limit.maxLimitRequestRatio[resourceName]}`}/>
+      ))
+      }
+      {limit.max &&
+      Object.keys(limit.max).filter(key => key === resource).map(resourceName => (
+        <Badge key={resourceName} label={`max: ${limit.max[resourceName]}`}/>
+      ))
+      }
+      {limit.min &&
+      Object.keys(limit.min).filter(key => key === resource).map(resourceName => (
+        <Badge key={resourceName} label={`min: ${limit.min[resourceName]}`}/>
+      ))
+      }
+    </div>
+  );
+}
+
+@observer
+export class LimitRangeDetails extends React.Component<Props> {
+  render() {
+    const { object: limitRange } = this.props;
+
+    if (!limitRange) return null;
+    const containerLimits = limitRange.getContainerLimits();
+    const podLimits = limitRange.getPodLimits();
+    const pvcLimits = limitRange.getPVCLimits();
+
+    return (
+      <div className="LimitRangeDetails">
+        <KubeObjectMeta object={limitRange}/>
+
+        {containerLimits.length > 0 &&
+        <DrawerItem name={`Container Limits`} labelsOnly>
+          <DrawerItem name={`CPU`} className="firstChild">
+            {
+              containerLimits.map(limit =>
+                renderLimits(limit, Resource.CPU)
+              )
+            }
+          </DrawerItem>
+          <DrawerItem name={`Memory`}>
+            {
+              containerLimits.map(limit =>
+                renderLimits(limit, Resource.MEMORY)
+              )
+            }
+          </DrawerItem>
+          <DrawerItem name={`Ephemeral Storage`}>
+            {
+              containerLimits.map(limit =>
+                renderLimits(limit, Resource.EPHEMERAL_STORAGE)
+              )
+            }
+          </DrawerItem>
+        </DrawerItem>
+        }
```

Comment:
The same sort of factoring out could be done here too. A pointer:

1. The sub drawer items can be produced with from an array of `Resource`'s being passed in. Plus remember to add the `"key"` prop.